src/services/events/utils/types.ts

```diff
@@ -0,0 +1,101 @@
+import { CartDto, Product, User } from '../../../interfaces'
+
+export type EventCallbackFunction<Payload = undefined> = (payload: Payload) => void
+
+export type GetTypeFromInterface<T> = T extends EventCallbackFunction<infer Payload>
```

Comment:
Dziwna ta nazwa, może lepiej coś w stylu `GetCallbackPayload`? Bo chyba właśnie to robi